### PR TITLE
[flutter_plugin_tools] Improve version check error handling

### PR DESCRIPTION
**Pull request by stuartmorgan**
Catches invalid CHANGELOG formats and logs useful error messages for them,
rather than throwing FormatExceptions.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran [the auto-formatter]. (Note that unlike the flutter/flutter repo, the flutter/plugins repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I [updated pubspec.yaml](https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates) with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test exempt.
- [x] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[the auto-formatter]: https://github.com/flutter/plugins/blob/master/script/tool/README.md#format-code

### CODE REVIEW

File: script/tool/lib/src/version_check_command.dart [hunk 1]
```diff
@@ -358,11 +359,19 @@ ${indentation}HTTP response: ${pubVersionFinderResponse.httpResponse.body}
...
 
-    final Version? fromChangeLog =
-        versionString == null ? null : Version.parse(versionString);
-    if (fromChangeLog == null) {
-      printError(
-          '${indentation}Cannot find version on the first line CHANGELOG.md');
+    if (versionString == null) {
+      printError('${indentation}Unable to find a version in CHANGELOG.md');
+      print('${indentation}The current version should be on a line starting '
+          ' with"## ", either on the first non-empty line or after a "## NEXT" '
```
Comment (ditman) [inline script/tool/lib/src/version_check_command.dart:365]:
```suggestion
          ' with "## ", either on the first non-empty line or after a "## NEXT" '
```

File: script/tool/test/version_check_command_test.dart [hunk 2]
```diff
@@ -514,6 +514,73 @@ void main() {
...
+        commandError = e;
+      });
+
+      expect(commandError, isA<ToolExit>());
+      expect(
+        output,
+        containsAllInOrder(<Matcher>[
+          contains('Unable to find a version in CHANGELOG.md'),
+          contains('The current version should be on a line starting with '
+              '"## ", either on the first non-empty line or after a "## NEXT" '
```
Comment (ditman) [inline script/tool/test/version_check_command_test.dart:548]: How's this test passing without the space before the "##" above?

**Review by ditman — APPROVED** [review]
LGTM!

File: script/tool/test/version_check_command_test.dart (hunk 2 above)
Comment (ditman) [reply script/tool/test/version_check_command_test.dart:548]: (OK, it's [not passing](https://cirrus-ci.com/task/4581919208767488?logs=main#L302-L323) :P)

Comment (stuartmorgan) [reply script/tool/test/version_check_command_test.dart:548]: I made a "cosmetic" line wrapping change just before pushing, and apparently didn't re-run tests.

File: script/tool/lib/src/version_check_command.dart [hunk 3]
```diff
@@ -358,11 +359,19 @@ ${indentation}HTTP response: ${pubVersionFinderResponse.httpResponse.body}
...
 
-    final Version? fromChangeLog =
-        versionString == null ? null : Version.parse(versionString);
-    if (fromChangeLog == null) {
-      printError(
-          '${indentation}Cannot find version on the first line CHANGELOG.md');
+    if (versionString == null) {
+      printError('${indentation}Unable to find a version in CHANGELOG.md');
+      print('${indentation}The current version should be on a line starting '
+          ' with "## ", either on the first non-empty line or after a "## NEXT" '
```
Comment (ditman) [inline script/tool/lib/src/version_check_command.dart:365]:
```suggestion
          'with "## ", either on the first non-empty line or after a "## NEXT" '
```

This was printing `starting  with` (double space), and breaking the test.

Comment (stuartmorgan) [reply script/tool/lib/src/version_check_command.dart]: Thanks; I was being lazy and not re-syncing and fixing locally. If this fails again I'll actually re-run locally...

**Pull request merged**